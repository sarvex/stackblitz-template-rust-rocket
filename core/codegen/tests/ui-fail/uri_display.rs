#[macro_use] extern crate rocket;

#[derive(UriDisplayQuery)]
struct Foo1;
//~^ ERROR not supported

#[derive(UriDisplayQuery)]
struct Foo2();
//~^ ERROR not supported

#[derive(UriDisplayQuery)]
enum Foo3 { }
//~^ ERROR not supported

#[derive(UriDisplayQuery)]
enum Foo4 {
    Variant,
    //~^ ERROR not supported
}

#[derive(UriDisplayQuery)]
struct Foo5(String, String);
//~^ ERROR exactly one

#[derive(UriDisplayQuery)]
struct Foo6 {
    #[form(field = 123)]
    //~^ ERROR invalid value: expected string
    field: String,
}

#[derive(UriDisplayPath)]
struct Foo7(String, usize);
//~^ ERROR exactly one

#[derive(UriDisplayPath)]
struct Foo8;
//~^ ERROR exactly one

#[derive(UriDisplayPath)]
enum Foo9 {  }
//~^ ERROR not supported

#[derive(UriDisplayPath)]
struct Foo10 {
//~^ ERROR not supported
    named: usize
}
