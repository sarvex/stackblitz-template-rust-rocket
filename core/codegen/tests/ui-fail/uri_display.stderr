error: fieldless structs or variants are not supported
 --> $DIR/uri_display.rs:4:1
  |
4 | struct Foo1;
  | ^^^^^^^^^^^^
  |
note: error occurred while deriving `UriDisplay`
 --> $DIR/uri_display.rs:3:10
  |
3 | #[derive(UriDisplayQuery)]
  |          ^^^^^^^^^^^^^^^

error: fieldless structs or variants are not supported
 --> $DIR/uri_display.rs:8:1
  |
8 | struct Foo2();
  | ^^^^^^^^^^^^^^
  |
note: error occurred while deriving `UriDisplay`
 --> $DIR/uri_display.rs:7:10
  |
7 | #[derive(UriDisplayQuery)]
  |          ^^^^^^^^^^^^^^^

error: empty enums are not supported
  --> $DIR/uri_display.rs:12:1
   |
12 | enum Foo3 { }
   | ^^^^^^^^^^^^^
   |
note: error occurred while deriving `UriDisplay`
  --> $DIR/uri_display.rs:11:10
   |
11 | #[derive(UriDisplayQuery)]
   |          ^^^^^^^^^^^^^^^

error: fieldless structs or variants are not supported
  --> $DIR/uri_display.rs:17:5
   |
17 |     Variant,
   |     ^^^^^^^
   |
note: error occurred while deriving `UriDisplay`
  --> $DIR/uri_display.rs:15:10
   |
15 | #[derive(UriDisplayQuery)]
   |          ^^^^^^^^^^^^^^^

error: tuple structs or variants must have exactly one field
  --> $DIR/uri_display.rs:22:12
   |
22 | struct Foo5(String, String);
   |            ^^^^^^^^^^^^^^^^
   |
note: error occurred while deriving `UriDisplay`
  --> $DIR/uri_display.rs:21:10
   |
21 | #[derive(UriDisplayQuery)]
   |          ^^^^^^^^^^^^^^^

error: invalid value: expected string literal
  --> $DIR/uri_display.rs:27:20
   |
27 |     #[form(field = 123)]
   |                    ^^^
   |
note: error occurred while deriving `UriDisplay`
  --> $DIR/uri_display.rs:25:10
   |
25 | #[derive(UriDisplayQuery)]
   |          ^^^^^^^^^^^^^^^

error: struct must have exactly one field
  --> $DIR/uri_display.rs:33:12
   |
33 | struct Foo7(String, usize);
   |            ^^^^^^^^^^^^^^^
   |
note: error occurred while deriving `UriDisplay`
  --> $DIR/uri_display.rs:32:10
   |
32 | #[derive(UriDisplayPath)]
   |          ^^^^^^^^^^^^^^

error: struct must have exactly one field
  --> $DIR/uri_display.rs:37:1
   |
37 | struct Foo8;
   | ^^^^^^^^^^^^
   |
note: error occurred while deriving `UriDisplay`
  --> $DIR/uri_display.rs:36:10
   |
36 | #[derive(UriDisplayPath)]
   |          ^^^^^^^^^^^^^^

error: enums are not supported
  --> $DIR/uri_display.rs:41:1
   |
41 | enum Foo9 {  }
   | ^^^^^^^^^^^^^^
   |
note: error occurred while deriving `UriDisplay`
  --> $DIR/uri_display.rs:40:10
   |
40 | #[derive(UriDisplayPath)]
   |          ^^^^^^^^^^^^^^

error: named structs are not supported
  --> $DIR/uri_display.rs:45:1
   |
45 | / struct Foo10 {
46 | | //~^ ERROR not supported
47 | |     named: usize
48 | | }
   | |_^
   |
note: error occurred while deriving `UriDisplay`
  --> $DIR/uri_display.rs:44:10
   |
44 | #[derive(UriDisplayPath)]
   |          ^^^^^^^^^^^^^^

error: aborting due to 10 previous errors

